src/bin/launch.rs

```diff
@@ -0,0 +1,82 @@
+use failure::bail;
+use structopt::StructOpt;
+use tsunami::providers::{Launcher, MachineSetup};
+use tsunami::TsunamiBuilder;
+
+#[derive(Debug)]
+enum Providers {
+    AWS,
+    Azure,
+}
+
+impl std::str::FromStr for Providers {
+    type Err = failure::Error;
+
+    fn from_str(s: &str) -> Result<Self, Self::Err> {
+        Ok(match s {
+            "aws" => Providers::AWS,
+            "azure" => Providers::Azure,
+            x => bail!("unknown provider {:?}", x),
+        })
+    }
+}
+
+#[derive(StructOpt)]
+struct Opt {
+    #[structopt(short = "p", long = "provider")]
+    provider: Providers,
+    #[structopt(short = "r", long = "region")]
+    region: String,
+}
+
+fn wait_for_continue(log: &slog::Logger) {
+    slog::debug!(
+        log,
+        "pausing for manual instance inspection, press enter to continue"
+    );
+
+    use std::io::prelude::*;
+    let stdin = std::io::stdin();
+    let mut iterator = stdin.lock().lines();
+    iterator.next().unwrap().unwrap();
+}
+
+fn launch_and_wait<M: MachineSetup + Clone, L: Launcher<MachineDescriptor = M>>(
+    b: TsunamiBuilder<M>,
+    l: &mut L,
+) -> Result<(), failure::Error> {
+    let log = b.logger();
+    b.spawn(l)?;
+    wait_for_continue(&log);
+    Ok(())
+}
+
+// just launch an instance in the specified region and wait.
+fn main() -> Result<(), failure::Error> {
```

Comment:
Is this generally a useful tool?

Reply:
Yeah, I have used it sometimes when I just want to launch a vm. Maybe that's just me though :)

Reply:
How about we move it into `examples/` instead?